dask-gateway-server/dask_gateway_server/backends/kubernetes/controller.py

```diff
@@ -1149,10 +1153,17 @@ def make_pod(self, namespace, cluster_name, config, is_worker=False):
         if extra_container_config:
             container = merge_json_objects(container, extra_container_config)
 
+        annotations = self.common_annotations
+        if extra_pod_annotations:
+            annotations = merge_json_objects(annotations, extra_pod_annotations)
+
+        if extra_pod_labels:
+            labels.update(extra_pod_labels)
```

Comment:
No need for conditional here, `extra_pod_labels` will always be a dict (it just might be empty).

Reply:
Since `labels` isn't used before then, you might also move https://github.com/dask/dask-gateway/blob/5377c03875682cea760c5ce61add8b646ec170aa/dask-gateway-server/dask_gateway_server/backends/kubernetes/controller.py#L1120 down to here to better group the logic.